### DISCUSSION TITLE
Data from dynamically generated checkboxes not saving to the database.

### DISCUSSION THREAD

**Discussion by ItsJoshDaly** [livewire/livewire#4137] [1. Help]
Hi there,

We're having an issue where we're trying to save the values of a checkbox to a jsonable column in our database. The data can be read and the checkboxes are displayed, but you can't see the values unless you dd() the element when you click save. 

```blade
<div class="bg-white">
    <div>
    @foreach($element->options as $checkbox_input)
        <label class="border-2 p-3 mr-1">
            <input wire:model="checkbox" type="checkbox" name="{{ $checkbox_input['title'] }}" value="{{ $checkbox_input['selected'] }}"> {{ $checkbox_input['title'] }}
        </label>
    @endforeach
    <button class="btn btn-approve" wire:click="saveButton">Save Changes</button>
    </div>
</div>
```

```php
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class CheckboxInput extends Component
{
    public $element;
    public $results;
    public $value;
    public $title;
    public $checkbox;

    public function saveButton() {

        $this->checkbox = $this->element->options;

        $collection = collect(['title' => $this->checkbox[0], 'selected' => $this->checkbox[1]]);

        $map = $collection->mapWithKeys(function ($item, $key) {
            return [$item["title"] => $item["selected"]];
        });

        $this->element->options = $map;
        dd($this->element->options);
//        $this->element->save();
    }

    public function render()
    {
        return view('livewire.checkbox-input')->with($this->results);
    }
}
```

I've not had much experience using collections for this kind of thing, so it might just be the case that the solution I'm using is also inherently wrong. Ideally, we'd have an output that looks like...

`[title => 'example', selected => 'true'], [title => 'example2, selected => 'false'], etc, etc`.

Any help would be muchly appreciated, thanks!

EDIT: Apologies for the weird formatting, some of the code in the format blocks would disappear if I used the proper formatting, bizarrely.

**Comment by joshhanley** [comment]
@ItsJoshDaly when you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

I have edited your post with the fix 🙂

**Reply by ItsJoshDaly** [reply]
Thanks, I was clicking the code shortcut on the comment editor and it would only give me two backticks (i.e. ``), but now I know how to sort it.

**Comment by joshhanley** [comment]
@ItsJoshDaly ok so there are a couple of issues, one is, you are overwriting `$this->checkbox` in your save method, so anything that would have been put in there by `wire:model="checkbox"` will now be gone.

So I would remove `$this->checkbox = $this->element->options;`

I would also remove this line `$collection = collect(['title' => $this->checkbox[0], 'selected' => $this->checkbox[1]]);`.

Ideally if you do a `dd($this->checkbox);` at the start of `saveButton`, you should see an array which has a list of titles that are selected, is that right?

If so, you should change your input to have the `title` as the value, like this

```blade
<input wire:model="checkbox" type="checkbox" value="{{ $checkbox_input['title'] }}" />
```

Then you should ensure your `$checkbox` property is initialised as an empty array `public $checkbox = [];`.

Making these changes with the `dd($this->checkbox);` at the top of save you should get the array as described above.

From there you can map through them and put the array into the correct format for saving into your model.

Hope this helps!

**Reply by ItsJoshDaly** [reply]
Hi there, I was actually getting a 1 or a 0 depending on whether or not it was selected with your suggested changes, but that's already getting us further along. What you suggest then is that we map the title and selected values in the back-end for the livewire component?

Thank you very much, by the way!

**Reply by joshhanley** [reply]
@ItsJoshDaly great! Yep exactly.

No worries 🙂
